Update README.md (#98)
floe/backscrub: README.md

![Screenshots with my stupid grinning face](images/screenshot.jpg)
(Credits for the nice backgrounds to [Mary Sabell](https://dribbble.com/shots/4686178-Bauhaus-Poster) and [PhotoFunia](https://photofunia.com/effects/retro-wave))

## Maintainers

  * Phil Ashby (@phlash)

>   * Benny Baumann (@benbe)

Reviewer: Preferred capitalization of the nickname is as follows … But all lower is acceptable too.
```suggestion
  * Benny Baumann (@BenBE)
```

  * Florian Echtler (@floe)

## Building

Install dependencies (`sudo apt install libopencv-dev build-essential v4l2loopback-dkms curl`).

Clone this repository with `git clone --recursive https://github.com/floe/backscrub.git`.
To speed up the checkout you can additionally pass `--depth=1` to `git clone`.
This is okay, if you only want to download and build the code, however, for development it is not recommended.

> Use `cmake` to build the project: create a subfolder (e.g. `build`), change to that folder and run: `cmake && make -j4` (please replace "4" with the number of your cores).

Reviewer 2: I usually work in the source folder with:
```bash
% cmake -B build
% cmake --build build
```
but that's personal pref/taste. If using the `cd` approach, then the parent directory needs to be referenced ie:
`cmake .. && make -j <n>`

Reviewer: ACK on the `cd` approach.

> Alternatively, you can also run `make` to build everything (should also clone and build Tensorflow Lite).

Reviewer 2: We may want to deprecate this now, it doesn't produce the same results (no XNNPACK), and Tensorflow have all but dropped support.

Reviewer: ```suggestion
**Deprecated**: Another option to build everything is to run `make` in the root directory of the repository. While this will download and build all dependencies, it comes with a few drawbacks like missing support for XNNPACK. Also this might break with newer versions of Tensorflow Lite as upstream support for this option has been removed. Use at you own risk.
```

## Usage

First, load the v4l2loopback module (extra settings needed to make Chrome work):
```

> sudo modprobe v4l2loopback devices=1 max_buffers=2 exclusive_caps=1 card_label="VirtualCam"

Reviewer 2: I like to force the device numbering here, avoids confusion:
```bash
sudo modprobe v4l2loopback video_nr=10 label="Virtual Cam" max_buffers=2 exclusive_caps=1
```

Reviewer: Maybe also document the configuration via `/etc/modprobe.d/v4l_loopback.conf` by setting up a configuration like:
```
# V4L loopback driver
options v4l2loopback max_buffers=2
options v4l2loopback exclusive_caps=1
# options v4l2loopback devices=4
options v4l2loopback card_label="VirtualCam 0,VirtualCam 1,VirtualCam 2,VirtualCam 3"
```

Doing so reduces the complexity to load the driver. Also adding `/etc/modules-load.d/v4l2loopback.conf` with the following content allows to auto-load the driver on every boot:
```
v4l2loopback
```

```
Then, run deepseg (-d -d for full debug, -c for capture device, -v for virtual device):
```
./deepseg -d -d -c /dev/video0 -v /dev/video1
```

> Some cameras like a *Logitec Brio* might need to switch the video source to `MJPG` by passing `-f MJPG` in order to allow for higher resolutions.

Reviewer: ```suggestion
Some cameras (like e.g. `Logitec Brio`) need to switch the video source to `MJPG` by passing `-f MJPG` in order for higher resolutions to become available for use.
```

## Requirements

Tested with the following dependencies:

  - Ubuntu 20.04, x86-64
    - Linux kernel 5.6 (stock package)
    - OpenCV 4.2.0 (stock package)
    - V4L2-Loopback 0.12.5 (stock package)
    - Tensorflow Lite 2.5.0 (from [repo](https://github.com/tensorflow/tensorflow/tree/v2.5.0/tensorflow/lite))
  - Ubuntu 18.04.5, x86-64
    - Linux kernel 4.15 (stock package)
    - OpenCV 3.2.0 (stock package)
    - V4L2-Loopback 0.10.0 (stock package)
    - Tensorflow Lite 2.1.0 (from [repo](https://github.com/tensorflow/tensorflow/tree/v2.1.0/tensorflow/lite))

>   

Reviewer 2: Can also add my Debian 10 (stable) with stock kernel, opencv, v4l2loopback and TFlite 2.5.0

Author: Please do :-)
